### PR TITLE
Support Mac Unit Tests

### PR DESCRIPTION
**Pull request by tomspilman**
This PR is a fix to support running Mac unit tests.

### CODE REVIEW

**Comment by tomspilman** [comment]
ugh... still something wonky with our develop branch adding these old commits that don't apply.

Still doesn't break this PR.

**Comment by tomspilman** [comment]
Woot... tests are running on the Mac

**Comment by Jjagg** [comment]
Awesome! Lots of failures though. But most are audio.

**Comment by tomspilman** [comment]
@KonajuGames @dellis1972 

Ok... audio tests passing, but we have a few others that still fail.

http://teamcity.monogame.net/viewLog.html?buildId=55098&tab=buildResultsDiv&buildTypeId=MonoGame_TestMac&guest=1

Let me know what needs to be done there and i'll fix them.  Or maybe we should work around them for now?

**Comment by Jjagg** [comment]
> should i be seeing the GameWindow popup on these tests

Nope, there's maybe 1 or 2 tests that actually create a game window, but IIRC they're ignored for DesktopGL.

@dellis1972 The content reference test is failing. It would be nice if it printed some more information. Maybe it can just print the Task Log if the test failed? This test also failed for me locally on Windows with MSBuild. I haven't really looked in to it yet, but anyway seems like it's not 100% right yet. 

About the IntermediateSerializer Colors test: it seems like it tries to parse the whiteline between two tags as content. Since it works on Windows, maybe it's an obscure Mono bug or something?

All other failures are about compression. They are failing because of DllNotFound for NVidia Texture Tools (nvtt). Could be the same issue as I had in #5866 or some other missing dependency. We really don't need nvtt btw (alternative with 0 deps in #5866).

So that's not too bad :D

**Comment by tomspilman** [comment]
So i think i'll just disable these last few failed tests and let someone else come in and fix them in a different PR.

**Comment by tomspilman** [comment]
So the current problem is that the unit tests hang at the end.  I suspect it is crashing, but i cannot be sure.  Needs more investigation.

**Comment by tomspilman** [comment]
@cra0zy @Jjagg @dellis1972 @KonajuGames -  This PR is ready to merge.

Note all other PRs will need to merge/rebase once this is committed else they will not pass the Mac unit tests.

Currently this is only running 36 of our 1200 tests on Mac because we're stuck on an old version of NUnit.  As soon as i work out the configuration error on the Mac build box that keeps the newer version of NUnit from working i can update the build server and all the tests will run.

Still for now lets move forward on this PR.

File: Test/ContentPipeline/BitmapContentTests.cs [hunk 1]
```diff
@@ -174,6 +174,9 @@ public void BitmapConvertFullNoResize()
         }
 
         [Test]
+#if DESKTOPGL
+        [Ignore("NVidia native library could not be loaded on Mac build server.")]
```
Comment (dellis1972) [inline Test/ContentPipeline/BitmapContentTests.cs]: @tomspilman what is the problem here? I had these working before.

Comment (tomspilman) [reply Test/ContentPipeline/BitmapContentTests.cs]: Don't know... exactly like i said it wasn't loading the library on the Mac build server.

However we have bigger issues to solve first as i cannot run the latest NUnit on the Mac keeping us from running 95% of our tests.

Comment (Jjagg) [reply Test/ContentPipeline/BitmapContentTests.cs]: These should work after a rebase/merge because we switched to the managed implementation in #6102.

Comment (tomspilman) [reply Test/ContentPipeline/BitmapContentTests.cs]: @Jjagg - But we're switching back to NVTT soon.  https://github.com/MonoGame/MonoGame/pull/6102#issuecomment-368181625

**Comment by tomspilman** [comment]
Bump... lets get this merged and i will work on fixing the mac build server to run all our tests this week.

**Pull request merged**